src/sql/workbench/contrib/dashboard/browser/core/dashboardPage.component.ts

```diff
@@ -495,14 +509,68 @@ export abstract class DashboardPage extends AngularDisposable implements IConfig
 		return tab.container ? Object.keys(tab.container)[0] : '';
 	}
 
-	private addNewTab(tab: TabConfig): void {
-		const existedTab = this.tabs.find(i => i.id === tab.id);
-		if (!existedTab) {
-			if (!tab.iconClass && tab.type !== 'group-header') {
-				tab.iconClass = 'default-tab-icon';
+	private async addNewTab(tab: TabConfig): Promise<void> {
+
+		try {
+			// Finding the owning extension for the tab
+			const owningExtension = this.extensionService.extensions.find(extension => {
+				if (extension?.contributes !== undefined && extension.contributes['dashboard.tabs']) {
+					const tabIndex = extension.contributes['dashboard.tabs'].findIndex(eTab => eTab.id === tab.id);
+					if (tabIndex !== -1) {
+						return true;
+					}
+				}
+				return false;
+			});
+
+			// If the tab is contributed by an extension, we'll wait for the extension to be activated before adding the tab.
+			// Not doing so causes issues with the tab UI not being rendered correctly.
+			if (owningExtension) {
+				// wait for the extension to be activated
+				await new Promise<void>((resolve, reject) => {
+					// If the extension is already activated, we can resolve immediately
+					if (this.extensionService.getExtensionsStatus()[owningExtension.id]?.activationTimes?.activateResolvedTime !== undefined) {
+						resolve();
+					}
+
+					// Otherwise, we'll wait for the extension to be activated
+					const disposable = this.extensionService.onDidChangeExtensionsStatus(e => {
+						e.forEach(extension => {
+							if (extension.value === owningExtension.id) {
+								disposable.dispose();
+								resolve();
+							}
+						})
+					});
+
+					/**
+					 * If the extension doesn't activate within 60 seconds, we'll reject the promise.
+					 * While this seems like a long time, it is possible for some extensions to take a
+					 * while to activate, especially if it has to download a big service file.
+					 */
+					setTimeout(() => {
+						disposable.dispose();
+						reject();
+					}, 60000);
+				});
 			}
-			this.tabs.push(tab);
-			this._cd.detectChanges();
+
+			const existedTab = this.tabs.find(i => i.id === tab.id);
+			if (!existedTab) {
+				if (!tab.iconClass && tab.type !== 'group-header') {
+					tab.iconClass = 'default-tab-icon';
+				}
+				this.tabs.push(tab);
+				this._cd.detectChanges();
+			}
+
+		} catch (error) {
+			// If we fail to load a tab, we'll just log the error and continue
+			this.notificationService.notify({
+				severity: Severity.Error,
+				message: nls.localize('dashboard.tabLoadingError', "Error loading tab {0}", tab.title)
```

Comment:
Not sure a timeout because an extension is taking a long time would count as an "error". Why do we even have a timeout anyways? Seems like it'd be fine just to not have that - and if an extension is stuck then it's stuck. But giving an error just because it's taking a while doesn't seem like the best experience.